src/hdf5_fun.cpp

```diff
@@ -1075,6 +1075,76 @@ hid_t
 
   }
 
+  static herr_t count_members( hid_t loc_id, const char *name, void *data ) {
+     (*(size_t*)data)++;
+     return 0;
+  }
+
+  BaseGDL* h5g_get_nmembers_fun( EnvT* e)
+  {
+    /* Dec 2022, Oliver Gressel <ogressel@gmail.com>
+    */
+    SizeT nParam=e->NParam(2);
+
+    hid_t h5f_id = hdf5_input_conversion(e, 0);
+
+    DString h5gGroupname;
+    e->AssureScalarPar<DStringGDL>( 1, h5gGroupname);
+
+    /* use 'H5Giterate()' to count members (as per IDL manual) */
+    size_t nmembers=0;
+    if ( H5Giterate( h5f_id, h5gGroupname.c_str(),
+                     NULL, count_members, &nmembers ) < 0 )
+       { string msg; e->Throw(hdf5_error_message(msg)); }
+
+    return new DLongGDL( nmembers );
+
+  }
+
+  static herr_t get_len( hid_t loc_id, const char *name, void *data ) {
+    static_cast<size_t*>(data)[0] = (name) ? strlen(name) : 0;
+    return 1;
+  }
+
+  static herr_t get_name( hid_t loc_id, const char *name, void *data ) {
+    strcpy(static_cast<char*>(data),name);
+    return 1;
+  }
+
+  BaseGDL* h5g_get_member_name_fun( EnvT* e)
+  {
+    /* Dec 2022, Oliver Gressel <ogressel@gmail.com>
+    */
+    SizeT nParam=e->NParam(3);
+
+    /* mandatory 'Loc_id' parameter */
+    hid_t loc_id = hdf5_input_conversion(e, 0);
+
+    /* mandatory 'Name' parameter */
+    DString name;
+    e->AssureScalarPar<DStringGDL>( 1, name);
+
+    /* mandatory 'Index' parameter */
+    DLong index;
+    e->AssureLongScalarKW(2, index);
+
+    /* use 'H5Giterate()' to get member name length */
+    size_t name_len;
+    if ( H5Giterate( loc_id, name.c_str(), &index, get_len, &name_len ) < 0 )
+       { string msg; e->Throw(hdf5_error_message(msg)); }
+    index--;
+
+    /* allocate string buffer (remains allocated) */
+    char* member_name=static_cast<char*>(calloc(name_len+1,sizeof(char)));
```

Comment:
Thanks @ogressel! One question: is it ensured that DStringGDL destructor will call free on the member_name? I'm trying to check it in `src/datatypes.hpp` (looking at `Data_`, looking for `free`) but cannot find an answer...

Reply:
Thanks for pointing this out. I adopted the idiom for creating a string object from a non-constant string from elsewhere, but was unaware that the "(remains allocated)" is in fact unintended behavior.